src/Duo.Domain/Trattamenti/Trattamento.cs

```diff
@@ -43,14 +44,33 @@ private void SetupCompleted()
         public void CambiaCodice(string codice)
         {
 
-            this.RaiseEvent<ITrattamentoCreato>(e =>
+            if (this.Data.Codice != codice)
             {
-                e.Codice = this.Data.Codice;
-                e.Descrizione = this.Data.Descrizione;
-            });
+                var vecchio = this.Data.Codice;
+                this.Data.Codice = codice;
+                this.RaiseEvent<ICodiceTrattamentoModificato>(e =>
+                {
+                    e.Vecchio = vecchio;
+                    e.Nuovo = this.Data.Codice;
+                });
+            }
 
         }
 
+        public void CambiaDescrizione(string descrizione)
+        {
+
+            if (this.Data.Descrizione != descrizione)
+            {
+                var vecchia = this.Data.Descrizione;
+                this.Data.Descrizione = descrizione;
+                this.RaiseEvent<IDescrizioneTrattamentoModificato>(e =>
+                {
+                    e.Vecchia = vecchia;
+                    e.Nuova = this.Data.Codice;
```

Comment:
Non è ` e.Nuova = this.Data.Codice;` ma credo debba essere ` e.Nuova = this.Data.Descrizione;`